update.go

```diff
@@ -0,0 +1,271 @@
+package plasmactlupdate
+
+import (
+	"errors"
+	"fmt"
+	"github.com/launchrctl/keyring"
+	"github.com/launchrctl/launchr/pkg/log"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+// Update stored update definition.
+type Update struct {
+	c        keyring.CredentialsItem
+	ext      string
+	fName    string
+	fTmpPath string
+	fPath    string
+}
+
+// CreateUpdate instance.
+func CreateUpdate(cr keyring.CredentialsItem) (*Update, error) {
+	return &Update{cr, "", "", "", ""}, nil
+}
+
+// Errors.
+var (
+	errUnsupportedOS   = errors.New("unsupported operating system")
+	errUnsupportedArch = errors.New("unsupported architecture")
+	errInvalidCreds    = errors.New("failed to validate credentials")
+)
+
+// Define the URL pattern for the file.
+const (
+	BaseUrl     = "https://repositories.skilld.cloud/repository/pla-plasmactl-raw"
+	releasePath = "stable_release"
+	binPathMask = "%s/%s/plasmactl_%s_%s%s"
+)
+
+// initVars initialize plugin variables.
+func (u *Update) initVars() (string, string, error) {
+	// Get username and password.
+	if err := u.getCreds(); err != nil {
+		return "", "", err
+	}
+
+	// Get the operating system type.
+	currOS, err := u.getOS()
+	if err != nil {
+		return "", "", err
+	}
+
+	// Set Update vars.
+	u.fName = fmt.Sprintf("plasmactl%s", u.ext)
+	u.fTmpPath = filepath.Join(os.TempDir(), u.fName)
+
+	// Get the machine architecture.
+	arch, err := getArch()
+	if err != nil {
+		return "", "", err
+	}
+
+	u.c.URL = fmt.Sprintf("%s/%s", BaseUrl, releasePath)
+	return currOS, arch, nil
+}
+
+// getCreds stores username and password credentials.
+func (u *Update) getCreds() error {
+	if u.c.Username != "" && u.c.Password != "" {
+		return nil
+	}
+
+	if u.c.URL != "" {
+		fmt.Printf("Please enter Username and Password for access to %s\n", u.c.URL)
```

Comment:
`"Please enter Username and Password for access to %s\n"` -> `"Enter credentials for %s\n"`